server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadReportAnalysisMetadataHolderStep.java

```diff
@@ -39,6 +44,12 @@ public LoadReportAnalysisMetadataHolderStep(BatchReportReader reportReader, Muta
   @Override
   public void execute() {
     BatchReport.Metadata reportMetadata = reportReader.readMetadata();
+    checkState(
+      ceTask.getComponentKey().equals(reportMetadata.getProjectKey()),
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'ceTask' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)